exporter/sumologicexporter/sender.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type appendResponse struct {
+	// sent gives information if the data was sent or not
+	sent bool
+	// appended keeps state of appending new log line to the body
+	appended bool
+	err      error
+}
+
+type sender struct {
+	buffer []pdata.LogRecord
+	config *Config
+	client *http.Client
+	filter filter
+	ctx    context.Context
```

Comment:
I'm missing some context here (no pun intended). Is this sender being created for each `pushDataLogs`, or is it a long-lived object? If it's long-lived, this `ctx` shouldn't be used for individual requests.

Reply:
It's going to be created for every `pushDataLogs`

More context: https://github.com/sumo-drosiek/opentelemetry-collector-contrib/blob/8e83f77433dcb5a6554b5538a769970cc66ab8aa/exporter/sumologicexporter/exporter.go#L104-L114